### DISCUSSION TITLE
Model lifecycle hooks context

### DISCUSSION THREAD

**Discussion by leonvor** [strapi/strapi#6920] [Legacy Topics]
Hi,

Are the model lifecycle hooks bound to the same request? Can I bind a value to the model to pass it between hooks in the same model, or is there a chance that multiple users can overwrite each other's request data? i.e Is it safe to do:

```
async beforeUpdate(params, data) {
  const {id} = params;
  const previous = await strapi.services.something.find({id: id});
  **this.previous = previous;**
},

afterUpdate(result, params, data) {
  // write to DB
  console.log(**this.previous**);
}

```
Where previous is the values of the item before update. I want to store values that changed in the update to a history table.
Thanks

**Comment by derrickmehaffy** [comment]
They are not directly no, AFAIK there is no real way to pass data between the hooks due to how they are triggered (maybe @alexandrebodin can correct me on this if I'm wrong).

Basically the flow that happens here:

User makes request => checked via users-permissions => route is checked (execute policies first) => execute controller => which calls the service => which calls the query => before query is ran the `beforeX` is ran => actual query gets executed => after query is ran the `afterX` is ran.

However, during the time when the beforeX or afterX happen nothing is passed along up or downstream, you aren't returning anything but simply performing a asynchronous action. The common usecase I see is say someone submits a "Contact us" form, after you save it to the database you trigger an email to some company contact email.

Or say a user is calling an endpoint that checks the weather, you want to cache that weather locally to cut down on remote API calls so in a `beforeX` you make a small find query and check the last `updated_at` value and if it's greater than xyz you make the remote API call and store it (probably a bad example there but you get the point).

Everything that happens within the lifecycle action is contained within that action.
